frontend/src/pages/TermPlanner/TermPlanner.tsx

```diff
@@ -34,11 +36,40 @@ const TermPlanner = () => {
   const planner = useSelector((state: RootState) => state.planner);
   const degree = useSelector((state: RootState) => state.degree);
 
-  const [termsOffered, setTermsOffered] = useState<Term[]>([]);
-  const [isDragging, setIsDragging] = useState(false);
+  const [draggingCourse, setDraggingCourse] = useState<string | undefined>();
 
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    // update the legacy term offered
+    const updateOfferingsData = async () => {
+      const years: string[] = [];
+      for (let i = 0; i < planner.numYears; i++) {
+        years.push((planner.startYear + i).toString());
+      }
+
+      // get the outdated legacy offerings
+      const outdated = Object.entries(planner.courses)
+        .filter(([, course]) => {
+          const oldOfferings = course.legacyOfferings;
+          // no legacy offering data, or doesn't include the wanted years
+          return !oldOfferings || years.filter((y) => !(y in oldOfferings)).length > 0;
```

Comment:
Ooo i like this approach! Doesn't need us to do a migration script for existing data already! :D

Reply:
<3